ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,225 @@
+        # TODO: solve credentials for clients
+        self.aws_client = S3Client()
+        self.google_client = GoogleClient()
+        self.azure_client = AzureClient()
+        self.s3comp_client = S3Client()
+
+
+class CloudClient(ABC):
+    """
+        Base abstract class for Cloud based Underlying Storage
+    """
+    client = None
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_uls(self, name):
+        """
+            Super method that first logs the ULS creation and then calls
+            the appropriate implementation
+        """
+        logger.info(f"Creating ULS: {name}")
+        self.internal_create_uls(name)
+
+    def delete_uls(self, name):
+        """
+            Super method that first logs the ULS deletion and then calls
+            the appropriate implementation
+        """
+        logger.info(f"Deleting ULS: {name}")
+        self.internal_delete_uls(name)
+
+    def get_all_uls_names(self):
+        pass
+
+    def verify_uls_exists(self, uls_name):
+        pass
+
+    @abstractmethod
+    def internal_create_uls(self, name):
+        pass
+
+    @abstractmethod
+    def internal_delete_uls(self, name):
+        pass
+
+
+class S3Client(CloudClient):
+    """
+        Implementation of a S3 Client using the S3 API
+    """
+
+    def __init__(self, key_id, access_key, endpoint="https://s3.amazonaws.com", verify=True,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.client = boto3.resource(
+            's3', verify=verify, endpoint_url=endpoint,
+            aws_access_key_id=key_id,
+            aws_secret_access_key=access_key
+        )
+        bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+        bs_secret_data['metadata']['name'] += f'-client-secret'
+        bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+        bs_secret_data['data']['AWS_ACCESS_KEY_ID'] = base64.urlsafe_b64encode(
+            self.aws_access_key_id.encode('UTF-8')
+        ).decode('ascii')
+        bs_secret_data['data']['AWS_SECRET_ACCESS_KEY'] = base64.urlsafe_b64encode(
+            self.aws_access_key.encode('UTF-8')
+        ).decode('ascii')
+        self.secret = create_resource(**bs_secret_data)
+
+    def internal_create_uls(self, name, region=None):
+        """
+            Creates the Underlying Storage using the S3 API
+        """
+        if region is None:
+            self.client.create_bucket(Bucket=name)
+        else:
+            self.client.create_bucket(
+                Bucket=name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+    def internal_delete_uls(self, name):
+        """
+            Deletes the Underlying Storage using the S3 API
+        """
+        self.client.meta.client.delete_bucket_policy(
+            Bucket=name
+        )
+        for _ in range(10):
```

Comment:
why 10?

Reply:
this is a copied function from an existing working test

Reply:
changed into TimeSampler